### PR TITLE
Add EndShipper beta class.

### PR DESCRIPTION
**Pull request by jchen293**
# Description

Adds the new EndShipper beta class with retrieve, all, and save functions. This is the fully-qualified address object that the USPS will start requiring shortly.
Add `BaseAddress` propTypes that `Address` and `EndShipper` can use

# Testing

Add unit tests

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement (fixing a typo, updating readme, renaming a variable name, etc)

### CODE REVIEW

File: src/beta/easypost.js [hunk 1]
```diff
@@ -1,10 +1,12 @@
 import Referral, { propTypes as referralPropTypes } from './resources/referral';
+import EndShipper from './resources/end_shipper';
```
Comment (Justintime50) [inline src/beta/easypost.js:2]: We need the propTypes for EndShipper here

File: src/beta/resources/end_shipper.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import T from 'proptypes';
+import base from '../../resources/base';
+
+export const propTypes = {
```
Comment (Justintime50) [inline src/beta/resources/end_shipper.js:4]: Are we able to consolidate and reuse this across `Address` and `EndShipper`?

File: test/resources/beta/end_shipper.test.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+  it('creates an EndShipper object', async function () {
+    const endShipper = await new this.easypost.EndShipper(Fixture.endShipperAddress()).save();
+
+    expect(endShipper).to.be.an.instanceOf(this.easypost.EndShipper);
+    expect(endShipper.id).to.match(/^es_/);
+    expect(endShipper.street1).to.equal('388 TOWNSEND ST APT 20');
+  });
+
+  it('retrieves an EndShipper', async function () {
```
Comment (Justintime50) [inline test/resources/beta/end_shipper.test.js:27]:
```suggestion
  it('retrieves an EndShipper object', async function () {
```

File: test/resources/beta/end_shipper.test.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    const endShipper = await new this.easypost.EndShipper(Fixture.endShipperAddress()).save();
+    const retrievedEndShipper = await this.easypost.EndShipper.retrieve(endShipper.id);
+
+    expect(retrievedEndShipper).to.be.an.instanceOf(this.easypost.EndShipper);
+    expect(endShipper.street1).to.equal(retrievedEndShipper.street1);
+  });
+
+  it('retrieves all EndShippers', async function () {
```
Comment (Justintime50) [inline test/resources/beta/end_shipper.test.js:35]:
```suggestion
  it('retrieves all EndShipper objects', async function () {
```

File: test/resources/beta/end_shipper.test.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    const endShippers = await this.easypost.EndShipper.all({ page_size: Fixture.pageSize() });
+    const endShippersArray = endShippers.end_shippers;
+
+    expect(endShippersArray.length).to.be.lessThanOrEqual(Fixture.pageSize());
+    endShippersArray.forEach((endShipper) => {
+      expect(endShipper).to.be.an.instanceOf(this.easypost.EndShipper);
+    });
+  });
+
+  it('updates an EndShipper', async function () {
```
Comment (Justintime50) [inline test/resources/beta/end_shipper.test.js:45]:
```suggestion
  it('updates an EndShipper object', async function () {
```

**Review by Justintime50 — CHANGES_REQUESTED** [review]
Let's figure out these propTypes before merging

File: src/beta/resources/end_shipper.js (hunk 2 above)
Comment (jchen293) [reply src/beta/resources/end_shipper.js]: yes, we can, added a `base_address` file

File: src/beta/resources/end_shipper.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import base from '../../resources/base';
+import { baseAddress as baseAddressPropTypes } from '../../resources/base_address';
+
+export const propTypes = Object.assign({}, baseAddressPropTypes);
```
Comment (Justintime50) [inline src/beta/resources/end_shipper.js:4]: We shouldn't need this line

File: src/beta/resources/end_shipper.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+export default (api) =>
+  class EndShipper extends base(api) {
+    static _name = 'EndShipper';
+
+    static _url = 'end_shippers';
+
+    static key = 'address';
+
+    static propTypes = propTypes;
```
Comment (Justintime50) [inline src/beta/resources/end_shipper.js:14]: We can directly assign `baseAddressPropTypes` here to `propTypes` probably.

File: src/beta/resources/end_shipper.js (hunk 6 above)
Comment (jchen293) [reply src/beta/resources/end_shipper.js:4]: I believe we need it for this line: https://github.com/EasyPost/easypost-node/blob/end_shipper/src/beta/easypost.js#L2

`{ propTypes as endShipperPropTypes }`

**Review by Justintime50 — APPROVED** [review]

**Review by nwithan8 — APPROVED** [review]

**Pull request merged**